spark/src/main/scala/org/apache/spark/sql/delta/commands/CloneTableBase.scala

```diff
@@ -314,7 +314,12 @@ abstract class CloneTableBase(
     clonedMetadata = clonedMetadata.copy(configuration =
       clonedMetadata.configuration.filterKeys(
         !CoordinatedCommitsUtils.TABLE_PROPERTY_KEYS.contains(_)).toMap)
-    clonedMetadata
+
+    val clonedSchema =
+      IdentityColumn.copySchemaWithMergedHighWaterMarks(
+        schemaToCopy = clonedMetadata.schema,
+        schemaWithHighWaterMarksToMerge = targetSnapshot.metadata.schema)
+    clonedMetadata.copy(schemaString = clonedSchema.json)
```

Comment:
We can combine the two `cloneMetadata.copy()` calls. We should make `cloneConfigurations` a separate variable for improved readability.